shell.nix

```diff
@@ -0,0 +1,5 @@
+with import <nixpkgs> {}; with haskellPackages;
+stdenv.mkDerivation {
+  name = "act";
+  buildInputs = [ BNFC alex happy ];
```

Comment:
What's `alex happy`?

Reply:
Two haskell packages that `bnfc` invokes in order to generate the parsers.
Alex is a lexical analyser generator https://www.haskell.org/alex/
Happy is a parser generator https://www.haskell.org/happy/